tests/functional/aws-node-sdk/test/object/mpuVersion.js

```diff
@@ -0,0 +1,748 @@
+        (res, uploadId, next) => {
+            const params = {
+                Bucket: bucketName,
+                Key: objectName,
+                MultipartUpload: {
+                    Parts: [
+                        {
+                            ETag: res.ETag,
+                            PartNumber: 1
+                        },
+                    ]
+                },
+                UploadId: uploadId,
+            };
+            const request = s3.completeMultipartUpload(params);
+            if (vId !== undefined) {
+                request.on('build', () => {
+                    request.httpRequest.headers['x-scal-s3-version-id'] = vId;
+                });
+            }
+            return request.send(err => next(err));
+        },
+    ], err => cb(err));
+}
+
+function putMPU(s3, bucketName, objectName, cb) {
+    return putMPUVersion(s3, bucketName, objectName, undefined, cb);
+}
+
+function checkVersionsAndUpdate(versionsBefore, versionsAfter, indexes) {
+    indexes.forEach(i => {
+        assert.notEqual(versionsAfter[i].value.Size, versionsBefore[i].value.Size);
+        assert.notEqual(versionsAfter[i].value.ETag, versionsBefore[i].value.ETag);
+        /* eslint-disable no-param-reassign */
+        versionsBefore[i].value.Size = versionsAfter[i].value.Size;
+        versionsBefore[i].value.ETag = versionsAfter[i].value.ETag;
+        /* eslint-enable no-param-reassign */
+    });
+}
+
+function checkObjMdAndUpdate(objMDBefore, objMDAfter, props) {
+    props.forEach(p => {
+        assert.notEqual(objMDAfter[p], objMDBefore[p]);
+        // eslint-disable-next-line no-param-reassign
+        objMDBefore[p] = objMDAfter[p];
+    });
+}
+
+describe('MPU with x-scal-s3-version-id header', () => {
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+
+        beforeEach(done => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return metadata.setup(() =>
+                s3.createBucket({ Bucket: bucketName }, err => {
+                    if (err) {
+                        assert.equal(err, null, 'Creating bucket: Expected success, ' +
+                            `got error ${JSON.stringify(err)}`);
+                    }
+                    done();
+                }));
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket');
+            return bucketUtil.empty(bucketName)
+            .then(() => {
+                process.stdout.write('Deleting bucket');
+                return bucketUtil.deleteOne(bucketName);
+            })
+            .catch(err => {
+                process.stdout.write('Error in afterEach');
+                throw err;
+            });
+        });
+
+        it('should overwrite an MPU object', done => {
+            let objMDBefore;
+            let objMDAfter;
+            let versionsBefore;
+            let versionsAfter;
+
+            async.waterfall([
```

Comment:
nitpick: If nothing is being `passed` on, may be use `series`.

Reply:
Yes, I will keep it in case we want to pass arguments later on.